### ISSUE TITLE
Sentry version 3.0.0-beta.4 breaks react native project imports in Android Studio

### ISSUE THREAD

**Issue by AlexanderEggers** [getsentry/sentry-android-gradle-plugin#283]
_Android Gradle Plugin:_
**7.0.4**

_Gradle:_
**7.3.3** 

_Optimizer:_
- [x] R8

_Have you created the sentry.properties file?_
- [x] Yes -> android folder of the react native project

_SDKs:_
- [x] sentry-react-native -> 3.2.13

---
I have the following issue:

I have an react native expo project setup and was trying to add the sentry gradle plugin (version 3.0.0-beta.4) to the rest of the project setup. Unfortunately I have noticed that this breaks the project integration with Android Studio. `yarn react-native run-android` still works fine, but looking at any classes in Android Studio (generated or other project files) none are able to properly import native modules provided by the react native autolinking. Additionally I have noticed that the kotlin plugin integration is broken again and is not functioning correctly anymore. Removing the sentry gradle plugin fixes the problems again.

**Steps to reproduce:**
- Run `create-react-native-app my-project` and create a project using expo.
- Update dependencies so that the mentioned versions for gradle are working fine, like using the latest RN version 0.67.2.
- Add `id "io.sentry.android.gradle" version "3.0.0-beta.4"` to the plugins block of the android/app/build.gradle file.
- Resync and rebuild the project
- After that is done, you will notice a lot of broken imports in your project. For example in your PackageList.java file which is generated by react native or in your Application.java.

**Actual result:**
- Imports provided via native modules from react-native autolinking are broken and do not resolve.

**Expected result:**
- Project can be used in Android Studio without problems.

**Comment by romtsn** [comment]
Hey, thanks for the report, 2 things:

1. > Additionally I have noticed that the kotlin plugin integration is broken again and is not functioning correctly anymore.

What do you mean by "again"? Is this a known problem that popped up in the past? Could you elaborate a bit more on the problem?

2. Could you try to `sentry.tracingInstrumentation.enabled = false` and see if that resolves the problems?

**Comment by AlexanderEggers** [comment]
> Hey, thanks for the report, 2 things:
> 
> 1. > Additionally I have noticed that the kotlin plugin integration is broken again and is not functioning correctly anymore.
> 
> What do you mean by "again"? Is this a known problem that popped up in the past? Could you elaborate a bit more on the problem?

Oh sorry I think that is a typo - had a long day :)

> 
> 2. Could you try to `sentry.tracingInstrumentation.enabled = false` and see if that resolves the problems?

Where do I add this part?

**Comment by romtsn** [comment]
In the same `build.gradle` file where you applied the sentry plugin

**Comment by AlexanderEggers** [comment]
@romtsn Thanks, that is working for me 👍  Are there any side-effects by using that?

**Comment by romtsn** [comment]
mm, don't think so, shouldn't be for react native. Unless you're using Room or File I/O on the android native site and want those to be automatically instrumented.

**Comment by AlexanderEggers** [comment]
No plan for that at the moment but are you planning to fix that behaviour with a later version of that gradle plugin?

**Comment by romtsn** [comment]
Yes, I think it's actually already fixed on `main`, I will cut a new release today and will ping you when it's available, so you could try again, if that's fine?

**Comment by AlexanderEggers** [comment]
Yep that works for me, thanks 👍

**Comment by cortinico** [comment]
> Yes, I think it's actually already fixed on `main`

@romtsn Out of curiosity, what was the issue?

**Comment by marandaneto** [comment]
> Yep that works for me, thanks 👍

Please try it out :) https://github.com/getsentry/sentry-android-gradle-plugin/releases/tag/3.0.0-rc.1

**Comment by romtsn** [comment]
@cortinico I'm hoping that this PR (#278) has fixed it, since we don't mess up with dependencies at configuration time anymore and using proper gradle hooks for that.

@AlexanderEggers could you give it a try with version `3.0.0-rc.1`?

**Comment by AlexanderEggers** [comment]
@romtsn @marandaneto Unfortunately the issue is still present in rc.1 when removing the code block you suggested earlier:

```
sentry {
    tracingInstrumentation {
        enabled = false
    }
}
```

**Comment by romtsn** [comment]
Just an update - I think I experienced the problem myself, and also found a fix for that, will test it a bit more and then supply a fix

**Comment by romtsn** [comment]
@marandaneto looks like this transform https://github.com/getsentry/sentry-android-gradle-plugin/blob/d8f1ab314ae113fef8c6404c9de39dd72d85c839/plugin-build/src/main/kotlin/io/sentry/android/gradle/transforms/MetaInfStripTransform.kt#L110-L142
doesn't work well with AGP's one, even though I tested it. The AGP's transforms probably do something that affects IDE index, and we intervene and therefore break something in that logic.

**Comment by marandaneto** [comment]
Did your workaround work? Do we have a fix?

**Comment by romtsn** [comment]
@AlexanderEggers could you give it a try with `3.0.0-rc.3` please?

**Comment by malbanese** [comment]
It looks like it's probably not completely fixed. I can work to get proper debugging details, but I ran into issues on 3.0.1. 

Downgrading to sentry gradle plugin version 2.1.5 does not exhibit the same problem.

It's possible that it runs into issues with Hilt's transformations, and whether it matters, the library it's failing on comes from Robolectric

Dagger / Hilt version: `2.41`
Hilt compiler version: `1.0.0`
Robolectric version: `4.6`
Sentry version: `5.7.3`
Sentry gradle version: `3.0.1`

```
* What went wrong:
Execution failed for task ':mobile:hiltAggregateDepsRedactedUnitTest'.
> Could not resolve all files for configuration ':mobile:hiltCompileOnlyRedactedUnitTest'.
   > Failed to transform bcprov-jdk15on-1.68-meta-inf-stripped.jar to match attributes {artifactType=aggregated-jar-for-hilt, meta-inf-stripped=true, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for MetaInfStripTransform: /Users/redacted/.gradle/caches/transforms-3/b6240e161c0c0d80ff310735f494e0cf/transformed/bcprov-jdk15on-1.68-meta-inf-stripped.jar.
         > Invalid signature file digest for Manifest main attributes
```

**Comment by romtsn** [comment]
@malbanese this looks like a different problem. Could you file a new issue please, with more details (e.g. which AGP version, etc.)? Thanks

**Comment by malbanese** [comment]
> @malbanese this looks like a different problem. Could you file a new issue please, with more details (e.g. which AGP version, etc.)? Thanks

@malbanese 

No problem, I was in the process of putting together a sample project demonstrating the error, and although the new version of the Sentry gradle plugin surfaced the build exception - it looks like it's actually an AGP 7.1.0 bug that got fixed in 7.1.2!

**Comment by romtsn** [comment]
@malbanese, yeah but this is because we don't run our transform on AGP 7.1.2 and above (so the issue is still there for the versions below). I will file a new issue 👍  thanks for reporting anyways.
